core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStore.java

```diff
@@ -362,7 +364,11 @@ public V put(K key, V value) throws StoreTimeoutException, StoreException {
     } catch (InterruptedException e) {
       throw new StoreException("Put operation interrupted while waiting for an ack from Kafka", e);
     } catch (ExecutionException e) {
-      throw new StoreException("Put operation failed while waiting for an ack from Kafka", e);
+      if (e.getCause() instanceof RecordTooLargeException) {
+        throw new EntryTooLargeException("Put operation failed because entry is too large");
```

Comment:
Do we need to make sure it works correctly with forwarding requests also?

Reply:
It should work as well as the other errors being returned.   We currently don't have tests that only send to a follower in a cluster.